### DISCUSSION TITLE
Help with formatting

### DISCUSSION THREAD

**Discussion by willemclarke** [helix-editor/helix#8745] [Troubleshooting] [answered]
Hi people, just posting a question:

I've recently started using helix and im loving it - but I'm having some trouble with using it at work as when i `:w` or `:fmt` the formatting is slightly off / touches pieces of code I didn't even interact with

![image](https://github.com/helix-editor/helix/assets/46552510/010f4013-3a5c-4c45-b5ab-a6191f5dd60a)
The following SS shows my general issue, the left is how it should be, and the right is what happens on `:w` / `:fmt`.

For context:
**vscode setup**
- Indent using spaces where tab size is **2**
- we use a `.prettierrc` file, which only sets `tabWidth: 2`
- rest of prettier settings within vscode is just defaults

**helix setup:**
`languages.toml`
```
[[language]]
name = "html"
formatter = { command = 'prettier', args = ["--parser", "html"] }

[[language]]
name = "json"
formatter = { command = 'prettier', args = ["--parser", "json"] }

[[language]]
name = "css"
formatter = { command = 'prettier', args = ["--parser", "css"] }

[[language]]
name = "javascript"
formatter = { command = 'prettier', args = ["--parser", "typescript"] }
auto-format = true

[[language]]
name = "typescript"
formatter = { command = 'prettier', args = ["--parser", "typescript"] }
auto-format = true

[[language]]
name = "tsx"
formatter = { command = 'prettier', args = ["--parser", "typescript"] }
auto-format = true
```
`config.toml`
```[editor]
auto-format = true
true-color = true
bufferline = "always"

[editor.cursor-shape]
insert = "bar"
normal = "block"
select = "underline"
```
when I run `:indent-style` I get 2 spaces

In my `languages.toml` I've tried setting  `indent = { tab-width = 2, unit = "  " }` / `indent = { tab-width = 2, unit = "    " }` for js, ts & tsx but to no avail

Any help would be much appreciated, cheers

**Comment by kirawi** [comment]
Indent-style only affects indentation within the editor, not formatting. Prettier probably has an arg somewhere to set indentation.

**Reply by willemclarke** [reply]
Oh okay thats good to know. Re prettier do you mean like this: (previously had this exact config for typescript, javascript, tsx - but didn't seem to help):
```
[[language]]
name = "javascript"
formatter = { command = 'prettier', args = ["--parser", "typescript", "--tab-width", "2"] }
auto-format = true
```
Looking through the options on prettier I'm not 100% sure which setting I need to add tbh

**Reply by willemclarke** [reply, marked as answer]
Just replying in-case anyone is as daft as me... I didn't have prettier installed locally on my machine - I assumed it was when I installed the prettier extension within vscode. Everything now works pleasantly.
